sympy/physics/units/prefixes.py

```diff
@@ -73,6 +73,9 @@ def __str__(self):
     __repr__ = __str__
 
     def __mul__(self, other):
+        if isinstance(sympify(other), Number):
+            return other*self
```

Comment:
This will ignore Python numbers (`Number` is a SymPy class).

Reply:
I would suggest something like 
```
if not isinstance(other, Prefix):
    super(Prefix, self).__mul__(other)
```

Reply:
After using `sympify(other)` all python numbers will get converted to sympy Numbers isn't it.
If I am missing something please point out.

Reply:
However making the changes suggested by you will make it work with symbols and other types as well.